### PR TITLE
Azure prow refactor

### PR DESCRIPTION
**Pull request by jim-minter**
```release-note
NONE
```

Remove ci-operator for all use cases except conformance and bushslicer

Counterpart to https://github.com/openshift/release/pull/3464

### CODE REVIEW

File: hack/verify/validate-codecov.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
```
Comment (asalkeld) [inline hack/verify/validate-codecov.sh]: I think the above way is the more correct way of doing it.

File: Makefile [hunk 2]
```diff
@@ -54,15 +50,18 @@ sync-run: generate
...
 	go build -ldflags ${LDFLAGS} ./cmd/$@
+.PHONY: monitoring
 
 monitoring-run: monitoring
 	./hack/monitoring.sh
 
 monitoring-stop:
 	./hack/monitoring.sh clean
 
+.PHONY: monitoring-run monitoring-stop
```
Comment (asalkeld) [inline Makefile]: the resultant Makefile .PHONY's look a bit of a mess. can you either place them after the target, in sections or at the top?

File: Makefile [hunk 3]
```diff
@@ -1,30 +1,25 @@
...
+	@rm -rf secrets
+	@mkdir secrets
+	@oc extract -n azure secret/cluster-secrets-azure --to=secrets >/dev/null
 
 clean:
 	rm -f coverage.out azure releasenotes
 
 generate:
-	go generate ./...
+	$(shell [[ -e /var/run/secrets/kubernetes.io ]] || go generate ./...)
```
Comment (ehashman) [inline Makefile]: Do we want to use `[` for shell compatibility?

File: Makefile (hunk 2 above)
Comment (ehashman) [reply Makefile]: +1, I'd vote for a single .PHONY target at the top.

Comment (jim-minter) [reply Makefile]: can do this as a follow-up - what I'm even more interested in is reducing the Makefile down even further...

**Comment by jim-minter** [comment]
> General comment: we're removing -x from a lot of these scripts, but I've found the output pretty helpful for debugging... what's the benefit of doing so? Less log volume?

The tradeoff is between mental load to read through lots of `-x` logs and risking not having logged the critical thing needed to debug.  Hopefully we can get to a point where the boilerplate test setup stuff is minimal and reliable enough that we don't need `-x` there.  I'm not against judicious logging, but IMO we'd gone too far.

**Comment by jim-minter** [comment]
tests passed

**Pull request merged**